app-android/src/main/java/com/ustadmobile/port/android/view/HolidayCalendarEditFragment.kt

```diff
@@ -145,4 +164,93 @@ class HolidayCalendarEditFragment() : UstadEditFragment<HolidayCalendar>(), Holi
     }
 
 
+}
+
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+fun HolidayCalendarEditScreen(
+    uiState: HolidayCalendarEditUiState,
+    onHolidayCalendarChange: (HolidayCalendar?) -> Unit = {},
+    onClickAddItem: () -> Unit = {},
+    onDeleteItemClick: (Holiday) -> Unit = {},
+    onItemClick: (Holiday?) -> Unit = {}
+){
+    Column (
+        modifier = Modifier
+            .padding(8.dp)
+            .verticalScroll(rememberScrollState()),
+        horizontalAlignment = Alignment.CenterHorizontally
+    ){
+
+        UstadTextEditField(
+            value = uiState.holidayCalendar?.umCalendarName ?: "",
+            label = stringResource(id = R.string.name),
+            enabled = uiState.fieldsEnabled,
+            onValueChange = {
+                onHolidayCalendarChange(uiState.holidayCalendar?.shallowCopy {
+                    umCalendarName = it
+                })
+            }
+        )
+
+        ListItem(
+            modifier = Modifier.clickable {
+                onClickAddItem()
+            },
+            icon = {
+                Icon(
+                    Icons.Filled.Add,
+                    contentDescription = null
+                )
+            },
+            text = { Text(stringResource(id = R.string.add_a_holiday)) }
+        )
+
+        uiState.holidayList?.forEach { holiday ->
+
+            val holidayStart = rememberFormattedDate(timeInMillis = holiday.holStartTime, timeZoneId = "UTC")
+            val holidayEnd = rememberFormattedDate(timeInMillis = holiday.holEndTime, timeZoneId = "UTC")
+
+            ListItem(
+                modifier = Modifier
```

Comment:
The holiday itself does not have an icon, but we need to make the text align (e.g. "Add a holiday" and "Eid" should align and start at the same position.

![spacer](https://user-images.githubusercontent.com/4280622/209866112-08302cc8-30f5-4e92-8ab4-9210fb0a177a.png)

You can do this by adding a spacer like this:

```
ListItem(
   icon = {
      Spacer(modifier = Modifier.width(24.dp))
   }
}
```